src/storage/hummock_sdk/src/time_travel.rs

```diff
@@ -300,47 +148,42 @@ impl From<(&HummockVersionDelta, &HashSet<CompactionGroupId>)> for IncompleteHum
             trivial_move: delta.trivial_move,
             new_table_watermarks: delta.new_table_watermarks.clone(),
             removed_table_ids: delta.removed_table_ids.clone(),
-            // TODO: optimization: strip table change log
+            // TODO: optimization: strip table change log based on select_group
             change_log_delta: delta
                 .change_log_delta
                 .iter()
-                .map(|(table_id, log_delta)| (*table_id, stripped_change_log_delta(log_delta)))
+                .map(|(table_id, log_delta)| (*table_id, PbChangeLogDelta::from(log_delta).into()))
                 .collect(),
             state_table_info_delta: delta.state_table_info_delta.clone(),
         }
     }
 }
 
-impl IncompleteHummockVersionDelta {
-    /// Resulted `SStableInfo` is incompelte.
-    pub fn to_protobuf(&self) -> PbHummockVersionDelta {
-        PbHummockVersionDelta {
-            id: self.id.0,
-            prev_id: self.prev_id.0,
-            group_deltas: self.group_deltas.clone(),
-            max_committed_epoch: self.max_committed_epoch,
-            safe_epoch: self.safe_epoch,
-            trivial_move: self.trivial_move,
-            new_table_watermarks: self
-                .new_table_watermarks
-                .iter()
-                .map(|(table_id, watermarks)| (table_id.table_id, watermarks.to_protobuf()))
-                .collect(),
-            removed_table_ids: self
-                .removed_table_ids
-                .iter()
-                .map(|table_id| table_id.table_id)
-                .collect(),
-            change_log_delta: self
-                .change_log_delta
-                .iter()
-                .map(|(table_id, log_delta)| (table_id.table_id, log_delta.into()))
-                .collect(),
-            state_table_info_delta: self
-                .state_table_info_delta
-                .iter()
-                .map(|(table_id, delta)| (table_id.table_id, *delta))
-                .collect(),
+pub struct SstableIdInVersion(HummockSstableId);
+
+impl From<&SstableIdInVersion> for PbSstableInfo {
+    fn from(sst_id: &SstableIdInVersion) -> Self {
```

Comment:
Conversion between `PbSstableInfo` and `SstableIdInVersion` is lossy. We should make sure it will not be misused.